modules/flash-script.nix

```
@@ -14,8 +14,16 @@ let
 
   updateFirmware = pkgs.writeShellApplication {
     name = "update-jetson-firmware";
-    runtimeInputs = [ pkgs.nvidia-jetpack.otaUtils ];
+    runtimeInputs = [ config.systemd.package pkgs.nvidia-jetpack.otaUtils ];
     text = ''
+      # If this script is not ran on real hardware, don't attempt to perform an
+      # update. This script could potentially run in a few places, for example
+      # in <nixpkgs/nixos/lib/make-disk-image.nix>.
+      if [[ $(systemd-detect-virt) != "none" ]]; then
```

> `systemd-detect-virt` exits with a return value of 0 (success) if a virtualization technology is detected, and non-zero (error) otherwise.

https://www.freedesktop.org/software/systemd/man/latest/systemd-detect-virt.html#Description

```suggestion
      if systemd-detect-virt --quiet; then
```